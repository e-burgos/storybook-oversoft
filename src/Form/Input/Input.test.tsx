import React from 'react';
import InputComponent from './Input.component';
import { render, screen, fireEvent } from '@testing-library/react';

test('Input Test', async () => {
  const onChangeMock = jest.fn();
  const view = render(<InputComponent type="text" label="Name" onChange={onChangeMock} value="value" />);
  expect(view).toBeDefined();
  fireEvent.change(screen.getByTestId('Name'), { target: { value: 'value2' } });
  expect(onChangeMock).toHaveBeenCalled();
  expect(onChangeMock.mock.calls[0][0]).toBe('value2');
  expect(view.container).toMatchSnapshot();
});
