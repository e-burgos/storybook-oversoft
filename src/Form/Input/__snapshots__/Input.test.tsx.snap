// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Test 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  color: var(--oversoft-grey-scale-dark-grey, #333);
  leading-trim: both;
  text-edge: cap;
  font-family: Rubik;
  font-size: 12px;
  font-style: normal;
  font-weight: 500;
  line-height: 20px;
}

.c2 {
  display: flex;
  height: 30px;
  width: 100%;
  justify-content: flex-end;
  align-items: center;
  gap: 8px;
  align-self: stretch;
  border-radius: 8px;
  border: 1px solid var(--oversoft-grey-scale-stroke-grey, rgba(198, 196, 196, 0.53));
  outline: none;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      Name
    </label>
    <input
      class="c2"
      data-testid="Name"
      type="text"
      value="value"
    />
  </div>
</div>
`;
